chromadb/execution/expression/operator.py

```diff
@@ -17,6 +18,13 @@ def version(self) -> RequestVersionContext:
         )
 
 
+@dataclass
+class SegmentScan(Scan):
+    knn_id: UUID
+    metadata_id: UUID
+    record_id: UUID
```

Comment:
This is a child class of `Scan`. In the distributed executor, we "populate" segment information to the `Scan` operator in a query plan by replacing it with a `SegmentScan`, right before sending it to the query node. This is not directly part of `Scan` because local executor does not need to know about segment information, which is handled by `MetadataSegment` and `VectorSegment`.